scrunch/datasets.py

```diff
@@ -880,6 +880,146 @@ def remove(self):
+        object.__setattr__(self, attr, value)
+
+    def __getattr__(self, attr):
+        if attr in self._METADATA_ATTRIBUTES:
+            return self.resource.metadata[attr]
+
+        if attr == 'filters':
+            # return a list of `Filters` instead of the filters expr on `body`
+            _filters = []
+            for obj in self.resource.filters:
+                f_url = obj['filter']
+                _filters.append(
+                    Filter(self.dataset.resource.filters.index[f_url]))
+            return _filters
+
+        if attr == 'variables':
+            # return a list of `Variables` instead of the where expr on `body`
+            _var_urls = []
+            _var_map = self.resource.where.args[0].map
+            for v in _var_map:
+                _var_urls.append(_var_map[v]['variable'])
+
+            return [
+                Variable(entity, self.dataset)
+                for url, entity in self.dataset._vars
+                if url in _var_urls
+            ]
+
+        # all other attributes not catched so far
+        if attr in self._ENTITY_ATTRIBUTES:
+            return self.resource[attr]
+        raise AttributeError('CrunchBox has no attribute %s' % attr)
+
+    def __repr__(self):
+        return "<CrunchBox: title='{}'; id='{}'>".format(
+            self.title, self.id)
+
+    def __str__(self):
+        return self.title
+
+    def remove(self):
+        self.dataset.resource.session.delete(self.url)
+
+    @property
+    def widget_url(self):
+        return self.WIDGET_URL.format(id=self.id)
+
+    @widget_url.setter
+    def widget_url(self, _):
+        """ prevent edits to the widget_url """
+        raise AttributeError("Can't edit 'widget_url' of a CrunchBox")
+
+    def iframe(self, logo=None, dimensions=None):
+        dimensions = dimensions or self.DIMENSIONS
+        widget_url = self.widget_url
+
+        if not isinstance(dimensions, dict):
+            raise TypeError('`dimensions` needs to be a dict')
+
+        def _figure(html):
+            return '<figure style="text-align:left;" class="content-list-'\
+                    'component image">\n' + '  {}\n'.format(html) + \
+                    '</figure>\n'
+
+        _iframe = (
+            '<iframe src="{widget_url}" width="{dimensions[width]}" '
+            'height="{dimensions[height]}" style="border: 1px solid #d3d3d3;">'
+            '</iframe>')
+
+        if logo:
+            _img = '<img src="{logo}" stype="height:auto; width:200px;'\
+                   ' margin-left:-4px"></img>'
+            _iframe = _figure(_img) + _iframe
+
+        elif self.title:
+            _div = '<div style="padding-bottom: 12px">\n'\
+                   '    <span style="font-size: 18px; color: #444444;'\
+                   ' line-height: 1;">' + self.title + '</span>\n'\
+                   '  </div>'
+            _iframe = _figure(_div) + _iframe
+
+        print(_iframe.format(**locals()))
```

Comment:
I would prefer having this return a string, and let the client decide if they want it printed out or do something else with it.

Now, this is a representation that is exclusive for the purposes of Yougov Daily right? I'm hesitant about adding html that is case specific, regarding the style tags, div styling, etc. Maybe we can provide the iframe part only with the style passed as a parameter and leave the rest for the Yougov Daily context?